cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractAssemblyOperator.java

```diff
@@ -72,7 +74,7 @@
      */
     protected AbstractAssemblyOperator(Vertx vertx, PlatformFeaturesAvailability pfa, String kind,
                                        CertManager certManager, PasswordGenerator passwordGenerator,
-                                       AbstractWatchableResourceOperator<C, T, L, D, R> resourceOperator,
+                                       AbstractWatchableResourceOperatorWithStatus<C, T, L, D, R> resourceOperator,
```

Comment:
Small nitpick, this sounds like the operator has a status, mybe something like:
```
AbstractWatchableStatedResourceOperator
```

Reply:
or `AbstractWatchableStatusedResourceOperator`
But statused is not a real word, but maybe it doesn't matter here?

Reply:
I would suggest "stateful", but that might be confusing in a different way. I think statused is fine.